tapa/graphir_conversion/utils.py

```
@@ -176,3 +180,49 @@ def get_task_graphir_parameters(task: Task) -> list[ModuleParameter]:
             )
         )
     return graphir_params
+
+
+def get_leaf_port_connection_mapping(
+    task_port: Port, task_module: Module, arg: str
+) -> dict[str, str]:
+    """Get leaf task port and slot port mapping.
+
+    Given a leaf task port and its arg, find all related ports in the task module based
+    on cat. Return a mapping from the leaf module port name to the connected slot port
+    name.
+    """
+    matching_ports = {}
+    if task_port.cat.is_scalar:
+        matching_ports[task_port.name] = arg
+
+    elif task_port.cat.is_istream:
+        for suffix in ISTREAM_SUFFIXES:
+            module_port = task_module.get_port_of(task_port.name, suffix)
+            matching_ports[module_port.name] = get_stream_port_name(arg, suffix)
+
+    elif task_port.cat.is_ostream:
+        for suffix in OSTREAM_SUFFIXES:
+            module_port = task_module.get_port_of(task_port.name, suffix)
+            matching_ports[module_port.name] = get_stream_port_name(arg, suffix)
+
+    elif task_port.cat.is_mmap:
+        for suffix in M_AXI_SUFFIXES:
+            m_axi_port_name = get_m_axi_port_name(task_port.name, suffix)
+            if m_axi_port_name in task_module.ports:
+                matching_ports[m_axi_port_name] = get_m_axi_port_name(arg, suffix)
```

Should we raise an error if `m_axi_port_name not in task_module.ports`?